src/Evolution/Systems/CurvedScalarWave/Constraints.cpp

```diff
@@ -28,28 +27,14 @@ void one_index_constraint(
         constraint,
     const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& d_psi,
     const tnsr::i<DataVector, SpatialDim, Frame::Inertial>& phi) {
-  if (get_size(get<0>(*constraint)) != get_size(get<0>(phi))) {
-    *constraint =
-        tnsr::i<DataVector, SpatialDim, Frame::Inertial>(get_size(get<0>(phi)));
-  }
-  // Declare iterators for d_psi and phi outside the for loop,
-  // because they are const but constraint is not
-  auto d_psi_it = d_psi.cbegin();
-  auto phi_it = phi.cbegin();
-
-  for (auto constraint_it = (*constraint).begin();
-       constraint_it != (*constraint).end();
-       ++constraint_it, (void)++d_psi_it, (void)++phi_it) {
-    *constraint_it = *d_psi_it - *phi_it;
-  }
+  destructive_resize_components(constraint, get<0>(phi).size());
+  *constraint = tenex::evaluate<ti::i>(d_psi(ti::i) - phi(ti::i));
```

Comment:
use the return-by-reference overload of `tenex::evaluate`? also in next commit

Reply:
thanks, done :)